app/js/app/directives/graph_sketcher/GraphSketcher.js

```diff
@@ -1,1087 +1,317 @@
-                                    if (f.getDist(current, curves[i].pts[j]) < MOUSE_DETECT_RADIUS) {
-                                        p.cursor(p.MOVE);
-                                        found = true;
-                                        break;
-                                    }
-                                }
-                            }
-                        }
-
-                        // attached symbols
-                        if (!found) {
-                            function loop1(knots) {
-                                if (found) {
-                                    return;
-                                }
-
-                                for (var j = 0; j < knots.length; j++) {
-                                    var knot = knots[j];
-                                    if (knot.symbol != undefined && isOverSymbol(current, knot.symbol)) {
-                                        p.cursor(p.MOVE);
-                                        found = true;
-                                        return;
-                                    }
-                                }
-                            }
-
-                            function loop2(knots) {
-                                if (found) {
-                                    return;
-                                }
-
-                                loop1(knots);
-
-                                for (var j = 0; j < knots.length; j++) {
-                                    var knot = knots[j];
-                                    if (knot.xSymbol != undefined && isOverSymbol(current, knot.xSymbol)) {
-                                        p.cursor(p.MOVE);
-                                        found = true;
-                                        return;
-                                    }
-                                    if (knot.ySymbol != undefined && isOverSymbol(current, knot.ySymbol)) {
-                                        p.cursor(p.MOVE);
-                                        found = true;
-                                        return;
-                                    }
-
-                                }
-                            }
-
-
-                            for (var i = 0; i < curves.length; i++) {
-                                var interX = curves[i]['interX'];
-                                loop1(interX);
-
-                                var interY = curves[i]['interY'];
-                                loop1(interY);
+                        let found = "notFound";
 
-                                var maxima = curves[i]['maxima'];
-                                loop2(maxima);
-
-                                var minima = curves[i]['minima'];
-                                loop2(minima);
-
-                                if (found) {
-                                    break;
-                                }
+                        if (found == "notFound") {
```

Comment:
We can have `found` be a boolean value rather than a string as string comparison is more susceptible to awkward typo bugs.